sklearn/cluster/tests/test_k_means.py

```diff
@@ -1276,3 +1276,18 @@ def test_predict_does_not_change_cluster_centers(is_sparse):
 
     y_pred2 = kmeans.predict(X)
     assert_array_equal(y_pred1, y_pred2)
+
+
+@pytest.mark.parametrize("Estimator", [KMeans, MiniBatchKMeans])
+@pytest.mark.parametrize("init", ["k-means++", "random"])
+def test_sample_weight_init(Estimator, init, global_random_seed):
+    """Check that sample weight is used during init."""
+    rng = np.random.RandomState(global_random_seed)
+    X, _ = make_blobs(
+        n_samples=200, n_features=10, centers=10, random_state=global_random_seed
+    )
+    sample_weight = rng.uniform(size=200)
+    kmeans = Estimator(random_state=global_random_seed, init=init)
+    clusters_weighted = kmeans.fit(X, sample_weight=sample_weight).cluster_centers_
+    clusters = kmeans.fit(X).cluster_centers_
+    assert (clusters_weighted != clusters).any()
```

Comment:
This is not really testing that ``sample_weight`` is used to initialize the centers because it's also used during all iterations of ``fit``. I think it would be better to test the ``_init_centroids`` method directly